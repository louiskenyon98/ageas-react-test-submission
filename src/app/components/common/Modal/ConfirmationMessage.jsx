import React from 'react';
import styles from '../../../../themes/style.scss';

export default class ConfirmationMessage extends React.PureComponent {
    render() {
        return (
            <React.Fragment>
                <h1>Is you sure fam?</h1>
                {this.props.body}
                <button className={`${styles.modalButton} ui inverted primary button`}
                        onClick={this.props.accept.callback}>{this.props.accept.label}
                </button>
            </React.Fragment>
        )
    }
}
