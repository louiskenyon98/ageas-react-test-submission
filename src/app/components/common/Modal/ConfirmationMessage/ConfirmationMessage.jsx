import React from 'react';
import styles from '../Modal.styles.scss';

export default class ConfirmationMessage extends React.PureComponent {
    render() {
        return (
            <React.Fragment>
                <h1>{this.props.title}</h1>
                {this.props.body}
                <div style={{float: 'right' , margin: 20}}>
                    {this.props.accept && <button
                        className={`${styles.modalButton} ui inverted primary button`}
                        onClick={this.props.accept.onClick}
                    >
                        {this.props.accept.label}
                    </button>}
                    {this.props.decline && <button
                        className={`${styles.modalButton} ui inverted red button`}
                        onClick={this.props.decline.onClick}
                    >
                        {this.props.decline.label}
                    </button>}
                </div>
            </React.Fragment>
        )
    }
}
