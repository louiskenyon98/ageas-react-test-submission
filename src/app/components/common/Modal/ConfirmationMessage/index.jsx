import React from 'react';
import PropTypes from 'prop-types';
import styles from '../../../../../themes/style.scss';

export default class ConfirmationMessage extends React.PureComponent {
    render() {
        return (
            <React.Fragment>
                <h1>{this.props.title}</h1>
                {this.props.body}
                <div style={{float: 'right' , margin: 20}}>
                    {this.props.decline && <button
                        className={`${styles.modalButton} ui inverted primary button`}
                        // onClick={this.props.accept.callback}>{this.props.accept.label}
                        onClick={this.props.decline.onClick}
                    >
                        {this.props.decline.label}
                    </button>}
                    {this.props.accept && <button
                        className={`${styles.modalButton} ui inverted primary button`}
                        // onClick={this.props.accept.callback}>{this.props.accept.label}
                        onClick={this.props.accept.onClick}
                    >
                        {this.props.accept.label}
                    </button>}
                </div>
            </React.Fragment>
        )
    }
}

ConfirmationMessage.propTypes = {
    title: PropTypes.string.isRequired,
    body: PropTypes.string,
    decline: PropTypes.shape({
        label: PropTypes.string.isRequired,
        onClick: PropTypes.func.isRequired
    }),
    accept: PropTypes.shape({
        label: PropTypes.string.isRequired,
        onClick: PropTypes.func.isRequired
    }),
};
