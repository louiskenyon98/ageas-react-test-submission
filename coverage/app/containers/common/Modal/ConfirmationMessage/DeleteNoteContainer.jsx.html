<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/containers/common/Modal/ConfirmationMessage/DeleteNoteContainer.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\..\index.html">All files</a> / <a href="index.html">app/containers/common/Modal/ConfirmationMessage</a> DeleteNoteContainer.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/6</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import {connect} from 'react-redux';
import {deleteNote, getNotes} from '../../../../actions/note/notes.action';
import {closeModal} from '../../../../actions/common/modal.action';
import DeleteNote from '../../../../components/common/Modal/ConfirmationMessage/DeleteNote';
&nbsp;
class DeleteNoteContainer extends React.Component {
<span class="fstat-no" title="function not covered" >    co</span>nstructor(props) {
<span class="cstat-no" title="statement not covered" >        super(props);</span>
<span class="cstat-no" title="statement not covered" >        this.deleteNote = this.deleteNote.bind(this);</span>
    }
&nbsp;
<span class="fstat-no" title="function not covered" >    de</span>leteNote() {
<span class="cstat-no" title="statement not covered" >        this.props.deleteNote(this.props.id, <span class="fstat-no" title="function not covered" >()</span> =&gt; {</span>
<span class="cstat-no" title="statement not covered" >            this.props.closeModal();</span>
<span class="cstat-no" title="statement not covered" >            this.props.getNotes();</span>
        })
    }
&nbsp;
<span class="fstat-no" title="function not covered" >    re</span>nder() {
<span class="cstat-no" title="statement not covered" >        return (</span>
            &lt;DeleteNote
                cancel={this.props.closeModal}
                delete={this.deleteNote}
            /&gt;
        )
    }
}
&nbsp;
export default connect(undefined, {deleteNote, getNotes, closeModal})(DeleteNoteContainer)
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Apr 18 2019 17:07:46 GMT+0100 (GMT Daylight Time)
</div>
</div>
<script src="..\..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\..\sorter.js"></script>
</body>
</html>
