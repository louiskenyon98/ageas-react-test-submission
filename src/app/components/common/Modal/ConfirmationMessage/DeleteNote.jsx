import React from 'react';
import ConfirmationMessage from "./ConfirmationMessage";

export default class DeleteNote extends React.PureComponent {
    render() {
        return (
            <ConfirmationMessage
                title={"Are you sure you want to delete your note"}
                body={"Hello world"}
                decline={{
                    label: "CANCEL",
                    onClick: this.props.cancel
                }}
                accept={{
                    label: "DELETE",
                    onClick: this.props.delete
                }}
            />
        )
    }
}
