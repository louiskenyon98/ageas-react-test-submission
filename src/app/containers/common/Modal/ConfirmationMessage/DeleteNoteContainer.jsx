import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {deleteNote, getNotes} from '../../../../actions/note/notes.action';
import {closeModal} from '../../../../actions/common/modal.action';
import DeleteNote from '../../../../components/common/Modal/ConfirmationMessage/DeleteNote';

export class DeleteNoteContainer extends React.Component {
    constructor(props) {
        super(props);
        this.deleteNote = this.deleteNote.bind(this);
    }

    deleteNote() {
        this.props.deleteNote(this.props.id, () => {
            this.props.closeModal();
            this.props.getNotes();
        })
    }

    render() {
        return (
            <DeleteNote
                cancel={this.props.closeModal}
                delete={this.deleteNote}
            />
        )
    }
}

DeleteNoteContainer.propTypes = {
  deleteNote: PropTypes.func,
  closeModal: PropTypes.func,
  getNotes: PropTypes.func
};

export default connect(undefined, {deleteNote, getNotes, closeModal})(DeleteNoteContainer)
