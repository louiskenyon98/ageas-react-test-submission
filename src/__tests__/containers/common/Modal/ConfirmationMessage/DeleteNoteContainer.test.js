import React from 'react';
import {shallow} from 'enzyme';
import {DeleteNoteContainer} from '../../../../../app/containers/common/Modal/ConfirmationMessage/DeleteNoteContainer';

describe('DeleteNoteContainer', () => {
    let props = {};
    let wrapper = () => shallow(<DeleteNoteContainer {...props}/>);
    beforeEach(() => {
        props = {
            id: 91,
            closeModal: jest.fn(),
            getNotes: jest.fn(),
            deleteNote: jest.fn((id, callback) => callback())
        }
    });
    describe('render', () => {
        it('should render DeleteNote component', () => {
            expect(wrapper()).toMatchSnapshot();
        })
    });
    describe('functionality', () => {
        describe('DeleteNote', () => {
            describe('props', () => {
                describe('cancel', () => {
                    it('should call closeModal', () => {
                        expect(props.closeModal).not.toHaveBeenCalled();
                        wrapper().find('DeleteNote').first().props().cancel();
                        expect(props.closeModal).toHaveBeenCalledTimes(1);
                    });
                });
                describe('delete', () => {
                    it('should call deleteNote (along with closeModal and getNotes)', () => {
                        expect(props.deleteNote).not.toHaveBeenCalled();
                        expect(props.closeModal).not.toHaveBeenCalled();
                        expect(props.getNotes).not.toHaveBeenCalled();
                        wrapper().find('DeleteNote').first().props().delete();
                        expect(props.deleteNote).toHaveBeenCalledTimes(1);
                        expect(props.closeModal).toHaveBeenCalledTimes(1);
                        expect(props.getNotes).toHaveBeenCalledTimes(1);
                    });
                });
            });
        })
    })
});
