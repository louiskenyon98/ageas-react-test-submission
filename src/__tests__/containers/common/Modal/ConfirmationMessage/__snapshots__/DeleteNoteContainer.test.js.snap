// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteNoteContainer render should render DeleteNote component 1`] = `
<DeleteNote
  cancel={[MockFunction]}
  delete={[Function]}
/>
`;
